symphony-app-authentication-fe/services/bootstrapService.js

```diff
@@ -0,0 +1,56 @@
+import { registerApplication } from './registerApplication';
+import {
+    authenticateApp,
+    validateTokens,
+    validateJwt
+  } from '../sagas/apiCalls';
+
+/*
+* initApp                                   initializes the communication with the Symphony Client
+* @params       config                      app settings
+* @return       SYMPHONY.remote.hello       returns a SYMPHONY remote hello service.
+*/
+export const initApp = (config) => {
+  let userInfo = {};
+  let tokenA = '';
+
+  // create our own service
+  SYMPHONY.services.register(`${config.appId}:controller`);
+
+  
+  const authenticateApplication = (podInfo) => {
+    return authenticateApp(podInfo.pod, config.appId);
+  }
+
+  const registerAuthenticatedApp = (appTokens) => {
+    tokenA = appTokens.data.appToken;
+
+    const appId = config.appId;
+    const appData = { appId, tokenA };
+
+    return registerApplication(config, appData);
+  }
+
+  const validateAppTokens = (symphonyToken) => {
+    return validateTokens(tokenA, symphonyToken.tokenS, config.appId);
+  }
+
+  const getJwt = () => {
+    const userInfoService = SYMPHONY.services.subscribe('extended-user-info');
+    return userInfoService.getJwt();
+  }  
+
+  const validateJwtToken = (jwt) => {
+    userInfo.jwt = jwt;
+    
+    return validateJwt(jwt);
+  }
+
+  SYMPHONY.remote.hello()
+  .then(authenticateApplication)
+  .then(validateAppTokens)
+  .then(validateJwtToken)
```

Comment:
You missed to cache the generated JWT locally. You should also provide an easy way to get this token once library consume will require this to perform API calls.